### PR TITLE
testbench: improvements for valgrind and debug

### PR DESCRIPTION
**Pull request by lgirdwood**
Add some improvements to support valgrind and eliminate all valgrind errors. Also allow users to specify number of copies.

### CODE REVIEW

File: tools/testbench/topology.c [hunk 1]
```diff
@@ -137,7 +137,7 @@ int load_buffer(void *dev, int comp_id, int pipeline_id,
 		struct snd_soc_tplg_dapm_widget *widget)
 {
 	struct sof *sof = (struct sof *)dev;
-	struct sof_ipc_buffer buffer;
+	struct sof_ipc_buffer buffer = {0};
```
Comment (marc-hb) [inline tools/testbench/topology.c:140]: Isn't the job of `tplg_load_buffer()` to initialize `buffer`? What error does this silence?

File: tools/testbench/common_test.c [hunk 2]
```diff
@@ -100,7 +100,7 @@ int tb_pipeline_params(struct ipc *ipc, struct pipeline *p,
 {
 	struct ipc_comp_dev *pcm_dev;
 	struct comp_dev *cd;
-	struct sof_ipc_pcm_params params;
+	struct sof_ipc_pcm_params params = {0};
```
Comment (marc-hb) [inline tools/testbench/common_test.c:135]: This looks like the initialization of some field(s) is missing below and this is merely hiding the issue.

File: tools/testbench/topology.c (hunk 1 above)
Comment (lgirdwood) [reply tools/testbench/topology.c:140]: testbench does not use all the data members as some are specific to HW config. This was only missing from the two "IPCs" in this PR.

File: tools/testbench/common_test.c (hunk 2 above)
Comment (lgirdwood) [reply tools/testbench/common_test.c:135]: ditto as above.

File: tools/testbench/topology.c (hunk 1 above)
Comment (marc-hb) [reply tools/testbench/topology.c:140]: I don't understand sorry, I don't have the full picture. What was the error reported and in which function?

Comment (lgirdwood) [reply tools/testbench/topology.c:140]: Use of uninitialized data in FW buffer logic due to testbench not initializing it. Not an issue with kernel as it clears. Testbench bug.

Comment (marc-hb) [reply tools/testbench/topology.c:140]: I understood that the error was testbench specific, my concern is that this change hides any "real" initialization bug in `tplg_load_buffer` in the future.

Comment (lgirdwood) [reply tools/testbench/topology.c:140]: Got you now, yep the userspace tplg parser is not setting any variables if it cant find tokens for them. I will fix this as another PR as it means removing more testbench code here and also testing other tools that used the shared tplg parser.

**Pull request merged**